### PR TITLE
HTCONDOR-1279 Vault credmon depends on python3-cryptography

### PR DESCRIPTION
**Pull request by jasoncpatton**
https://opensciencegrid.atlassian.net/browse/HTCONDOR-1279

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by timtheisen** [comment]
@jasoncpatton This may break AmazonLinux 2. It reports as RHEL7. However, it has Python 3.7. We need to make sure it works there as well.

**Comment by jasoncpatton** [comment]
> @jasoncpatton This may break AmazonLinux 2. It reports as RHEL7. However, it has Python 3.7. We need to make sure it works there as well.

Ah, bummer, I'm guessing it's going to fail there. We could still probably at least improve EL8+ distros by doing `python3-cryptography` `%if 0%{?rhel} > 7`

File: build/packaging/srpm/condor.spec [hunk 1]
```diff
@@ -572,6 +572,11 @@ Group: Applications/System
 Requires: %name = %version-%release
 Requires: python3-condor = %{version}-%{release}
 Requires: python3-six
+%if 0%{?rhel} == 7
```
Comment (timtheisen) [inline build/packaging/srpm/condor.spec:575]: Let make it work for RHEL 7, not Amazon Linux 2, and RHEL > 8.
```suggestion
if 0%{?rhel} == 7 && ! 0%{?amzn}
```

File: build/packaging/srpm/condor.spec [hunk 2]
```diff
@@ -572,6 +572,11 @@ Group: Applications/System
...
+Requires: python36-cryptography
+%else
```
Comment (timtheisen) [inline build/packaging/srpm/condor.spec:577]:
```suggestion
%endif
%if 0%{?rhel} >= 8
```

**Review by timtheisen — CHANGES_REQUESTED** [review]
Let's make it work for EL7 and ingore Amazon Linux 2.

Apply the suggested changes.

**Comment by jasoncpatton** [comment]
> Let's make it work for EL7 and ingore Amazon Linux 2.
> 
> Apply the suggested changes.

Done, thank you!

**Review by timtheisen — APPROVED** [review]

**Pull request merged**